### ISSUE TITLE
NodeResize add nodes test flaked

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#17765]
Sometimes it's not able to create a new working node quickly enough:
```
00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-i482 condition 1/2: type: Ready, status: False, reason: "KubeletNotReady", message: "ConfigureCBR0 requested, but PodCIDR not set. Will not configure CBR0 right now", last transition time: 2015-11-25 00:24:04 -0800 PST
00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-i482 condition 2/2: type: OutOfDisk, status: False, reason: "KubeletHasSufficientDisk", message: "kubelet has sufficient disk space available", last transition time: 2015-11-25 00:24:04 -0800 PST
00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-lthy condition 1/2: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-11-24 21:56:05 -0800 PST
00:24:10 Nov 25 00:24:10.029: INFO: Successfully found node e2e-slow-minion-lthy readiness to be true
00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-w470 condition 1/2: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-11-24 23:12:30 -0800 PST
00:24:10 Nov 25 00:24:10.029: INFO: Successfully found node e2e-slow-minion-w470 readiness to be true
00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-z1um condition 1/2: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-11-24 21:56:06 -0800 PST
00:24:10 Nov 25 00:24:10.029: INFO: Successfully found node e2e-slow-minion-z1um readiness to be true
00:24:10 Nov 25 00:24:10.029: INFO: Waiting for cluster size 4, current size 4, not ready nodes 1
```

**Comment by wojtek-t** [comment]
@gmarek - can you please provide a link to example failure?

**Comment by gmarek** [comment]
e.g. http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/971

**Comment by wojtek-t** [comment]
@kubernetes/goog-node - can someone please take a look into it?

**Comment by lavalamp** [comment]
@dchen1107 to delegate?

**Comment by dchen1107** [comment]
Shouldn't the logging message of 
```
"00:24:10 Nov 25 00:24:10.029: INFO: Node e2e-slow-minion-i482 condition 1/2: type: Ready, status: False, reason: "KubeletNotReady", message: "ConfigureCBR0 requested, but PodCIDR not set. Will not configure CBR0 right now", last transition time: 2015-11-25 00:24:04 -0800 PST"
``` 
clearly indicate the reason? Node  e2e-slow-minion-i482 is NotReady because network configuration is not done yet. But network configuration is not done is due to PodCIDR is not set because node controller didn't assign node CIDR to  e2e-slow-minion-i482.

**Comment by dchen1107** [comment]
@wojtek-t I am reassiging this to you since you are in charge of node controller today. Thanks!

**Comment by wojtek-t** [comment]
@gmarek owns node-controller - he already reassigned it to him

**Comment by gmarek** [comment]
It's not on the node-controllers side:
```
I1125 08:24:08.295936       6 nodecontroller.go:488] Assigning node e2e-slow-minion-i482 CIDR 10.245.3.0/24
I1125 08:24:13.416215       6 nodecontroller.go:477] CIDR 10.245.3.0/24 is already being used by node e2e-slow-minion-i482
```

Note that second line is printed only after Listing all nodes from API server. I.e. at 8:24:13 entry in master already had information about CIDR. Sadly we don't save logs from Nodes created in the autoscaling tests, which would make debugging a lot easier @fejta @dchen1107.

**Comment by gmarek** [comment]
I looked at everything that's available and everything looks good to me. Code also seems correct, but I'm not very familiar in this path in Kubelet. @bprashanth was doing something around CIDRs when working on Flannel - maybe he'll have some ideas.

Reassigning to @fejta, as I doubt we'll get far without logs from 'added' Node.

**Comment by krousey** [comment]
@gmarek assigning to you to handle or delegate to someone who can.

**Comment by lavalamp** [comment]
This happened again in this run: http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-parallel/14299/

**Comment by lavalamp** [comment]
Also seems to be killing the reboot test? http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gke-ci-reboot/6285

**Comment by gmarek** [comment]
@lavalamp - do you have any idea who has any knowledge of this? @kubernetes/goog-node or @kubernetes/goog-cluster? As I wrote earlier it doesn't look like a problem on NodeController side, so I guess it's Node. The main problem with this is that we don't have any logs from Nodes that are created and removed during restart test (cc @ixdy)

**Comment by yujuhong** [comment]
> Also seems to be killing the reboot test? http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gke-ci-reboot/6285

gke-ci-reboot started failing at build #6283. PR  #19513 was merged in that build which added at least 25~30 minutes of sleep time for each build (obtained by grepping `waiting 5 minutes for all dead tunnels to be dropped` from the logs). This might have just tipped gke-ci-reboot over the 120 minutes timeout.

**Comment by lavalamp** [comment]
@cjcullen or @ihmccreery, considering the above, can one of you take this?

**Comment by lavalamp** [comment]
Actually I guess #19540 is supposed to fix this?

**Comment by cjcullen** [comment]
I'd vote for increasing the overall timeout until we get #19314 in. We'll still have a post-reboot sleep in GKE, but it'll be 1 minute instead of 5.

**Comment by ihmccreery** [comment]
Yeah, I don't think this is related to the timeouts on `gke-ci-reboot` (see #19539).  I think this is a node team thing, but seems blocked on #18432.

**Comment by gmarek** [comment]
It may be the problem somewhere in client which may cause to silently drop the update somewhere.

**Comment by davidopp** [comment]
@dchen1107 I notice that you assigned this to me. But the two theories that I see presented above are:
(1) it's a node problem, or
(2) it's a CSI problem

Maybe you and @lavalamp can sort out who should take it?

**Comment by lavalamp** [comment]
This is blocked on #18432, which is an unstaffed test infrastructure request.

Basically we need logs from the kubelet, and they're not currently being stored because the node doesn't remain for the duration of the test.

**Comment by davidopp** [comment]
@lavalamp Any suggestion on what to do short-term? Move it to flaky test suite? Or just accept that we're not going to be able to get the number of flaky tests (in non-flaky suites) down to zero? I'm fine with either.

**Comment by spxtr** [comment]
Would calling `CoreDump` at the start and deferred end of `resizeGroup` allow you to debug this? It's not beautiful, but there's really not an easy way to do this in the short-term.

**Comment by lavalamp** [comment]
@spxtr That SGTM

**Comment by davidopp** [comment]
Who is going to implement that?

**Comment by spxtr** [comment]
I'm testing it out right now.

**Comment by davidopp** [comment]
Closing, please reopen when we see the problem again and we'll look at the new debug info that @spxtr added.
